server/src-lib/Hasura/Server/Migrate.hs

```diff
@@ -164,7 +164,9 @@ migrateCatalog env migrationTime = do
     buildCacheAndRecreateSystemMetadata :: m (RebuildableSchemaCache m)
     buildCacheAndRecreateSystemMetadata = do
       schemaCache <- buildRebuildableSchemaCache env
-      view _2 <$> runCacheRWT schemaCache recreateSystemMetadata
+      case scInconsistentObjs (lastBuiltSchemaCache schemaCache) of
+        [] -> view _2 <$> runCacheRWT schemaCache recreateSystemMetadata
+        _  -> pure schemaCache
```

Comment:
I'm not sure if this is the right solution. I tried to replicate the behaviour of when `previousVersion == latestCatalogVersionString` in the `migrateFrom` function in the `migrateCatalog` function. 

Calling the `runCacheRWT` function with a `schemaCache` containing inconsistent metadata throws an error, I don't understand the `runCacheRWT` function well enough to know why that happens.

@lexi-lambda Please do let me know if there is a better way to solve this issue 😄

Reply:
This is definitely *not* the right solution: we can’t just skip updating the system metadata if there are inconsistent objects! Then the system metadata might be wrong/out of date, and it wouldn’t be fixed until the next migration.

The reason an exception is raised is that `recreateSystemMetadata` calls `buildSchemaCacheStrict`, which raises an error on inconsistent metadata. Fortunately, this has a simple solution: call `buildSchemaCache` instead, which does not raise an error on inconsistent metadata.

Reply:
Oh, I didn't know that's what `buildSchemaCache` is used for 😅 . I have fixed this. Thanks!

Reply:
No worries! No harm no foul, and the updated patch looks good to me.